extensions/ilovepdf/src/add-page-number.tsx

```diff
@@ -54,74 +57,111 @@ export default function Command() {
   const [font, setFont] = useState<string>("Arial Unicode MS");
   const [fontSize, setFontSize] = useState<string>("14");
 
+  const [selectedFiles, setSelectedFiles] = useState<string[]>([]);
+
   async function handleSubmit(values: Values) {
     setIsLoading(true);
-    if (!values.files.length) {
+    if (!selectFileInFinder && !values.files.length) {
       await showToast(Toast.Style.Failure, "You must select a single file.", "Please select a file.");
       setStatus("failure");
       setIsLoading(false);
       return;
+    } else {
+      values.files = selectedFiles;
     }
 
     const toast = await showToast(Toast.Style.Animated, "Processing", "Adding Page Number...");
 
-    const file: string = values.files[0];
-    const fileExtension = path.extname(file);
-    const fileName = path.basename(file, fileExtension);
-    const directory = path.dirname(file);
-    let destinationFile = getFilePath(directory, `${fileName}_numbered.pdf`);
-
-    if (askBeforeDownload) {
-      const finalName = await chooseDownloadLocation(
-        destinationFile,
-        "Save The PDF As",
-        setIsLoading,
-        setStatus,
-        toast,
-      );
-      if (finalName == undefined) {
-        return;
+    values.files.map(async (valueFile) => {
```

Comment:
I have edited the code so that when there is a file failure, it will stop the process and inform the user about the error that occurred with the file.